node/src/components/block_synchronizer/global_state_synchronizer.rs

```diff
@@ -378,33 +461,48 @@ impl GlobalStateSynchronizer {
                 trie_hash,
                 trie_raw: *trie_raw,
                 request_root_hashes,
-                put_trie_result,
+                put_trie_result: put_trie_result.map(TrieHash),
             })
     }
 
-    pub(super) fn cancel_request(&mut self, root_hash: Digest, error: Error) -> Effects<Event> {
+    pub(super) fn cancel_request(&mut self, root_hash: RootHash, error: Error) -> Effects<Event> {
         match self.request_states.remove(&root_hash) {
-            Some(request_state) => request_state.respond(Err(error)),
-            None => Effects::new(),
+            Some(request_state) => {
+                debug!(%root_hash, "cancelling request");
+                self.fetch_queue.handle_request_cancelled(&root_hash);
+                // remove the association with the root hash in tries_awaiting_children
+                for trie_awaiting_children in self.tries_awaiting_children.values_mut() {
+                    trie_awaiting_children.handle_request_cancelled(&root_hash);
+                }
+                request_state.respond(Err(error))
+            }
+            None => {
+                debug!(%root_hash, "not cancelling request - root hash not found");
+                Effects::new()
+            }
         }
     }
 
-    fn finish_request(&mut self, trie_hash: Digest) -> Effects<Event> {
-        match self.request_states.remove(&trie_hash) {
+    fn finish_request(&mut self, root_hash: RootHash) -> Effects<Event> {
+        match self.request_states.remove(&root_hash) {
             Some(request_state) => {
+                debug!(%root_hash, "finishing request");
                 let unreliable_peers = request_state.unreliable_peers.iter().copied().collect();
-                request_state.respond(Ok(Response::new(trie_hash, unreliable_peers)))
+                request_state.respond(Ok(Response::new(root_hash, unreliable_peers)))
+            }
+            None => {
+                warn!(%root_hash, "not finishing request - root hash not found");
```

Comment:
During previous tests this has been happening a lot (more than one million logs over the course of 6 hrs).

Maybe it should stay as `debug` or `trace`, unless the updated flow prevents this from happening that often.

Reply:
It shouldn't be happening anymore, that's why I upgraded it to `warn` - if you look in `handle_trie_written`, now we check if the hash of the written trie is among the root hashes that have been requested and only call `finish_request` if it is. So if it is among the root hashes, but we don't have a corresponding request, that is weird and warrants a `warn` in my opinion.

Reply:
ok, thanks for explanation :+1: 
Indeed, that could even be an `error`, to be consistent with, for example, [this entry](https://github.com/casper-network/casper-node/blob/dev/node/src/components/block_synchronizer/trie_accumulator.rs#L169).

Reply:
Changed into `error` in 3a1521b